lib/agent/index.js

```diff
@@ -94,10 +94,10 @@ class Agent extends EventEmitter {
       return client.dispatch(opts, handler)
     }
 
+    /* istanbul ignore next */
```

Comment:
This can happen? It's just the data event that we don't bother testing for.

Reply:
If you look at line 90, there is an entire check about body  empty stream. So, the if on 98 is only satisfied by 
I was not able to simulate a empty stream which emits so I skipped the if check as well.

Reply:
I guess it doesn't matter so much.